From dd9c565466a72820846ff49f461b240cbb8b039e Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Thu, 20 Aug 2026 11:40:01 -0700
Subject: [PATCH] giflib files does not build with c-extension

pytorch_vision registers the bundled giflib sources
as CppExtension files in setup.py. This causes the
PyTorch builder to add the -std=c++17 build flag.

That flag causes build failures on at least macOS,
Fedora 40, and Mageia 9. Compile the bundled giflib
sources as C++ to avoid those failures.

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 setup.py                                                        | 2 +-
 .../csrc/io/image/cpu/giflib/{dgif_lib.c => dgif_lib.cpp}       | 0
 .../csrc/io/image/cpu/giflib/{gif_hash.c => gif_hash.cpp}       | 0
 .../csrc/io/image/cpu/giflib/{gifalloc.c => gifalloc.cpp}       | 0
 .../giflib/{openbsd-reallocarray.c => openbsd-reallocarray.cpp} | 0
 5 files changed, 1 insertion(+), 1 deletion(-)
 rename torchvision/csrc/io/image/cpu/giflib/{dgif_lib.c => dgif_lib.cpp} (100%)
 rename torchvision/csrc/io/image/cpu/giflib/{gif_hash.c => gif_hash.cpp} (100%)
 rename torchvision/csrc/io/image/cpu/giflib/{gifalloc.c => gifalloc.cpp} (100%)
 rename torchvision/csrc/io/image/cpu/giflib/{openbsd-reallocarray.c => openbsd-reallocarray.cpp} (100%)

diff --git a/setup.py b/setup.py
index 6568d25..914e948 100644
--- a/setup.py
+++ b/setup.py
@@ -401,7 +401,7 @@ def make_image_stable_extension():
     sources = (
         list(image_dir.glob("*.cpp"))
         + list(image_dir.glob("cpu/*.cpp"))
-        + list(image_dir.glob("cpu/giflib/*.c"))
+        + list(image_dir.glob("cpu/giflib/*.cpp"))
         + list(image_dir.glob("cuda/*.cpp"))
     )
 
diff --git a/torchvision/csrc/io/image/cpu/giflib/dgif_lib.c b/torchvision/csrc/io/image/cpu/giflib/dgif_lib.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/dgif_lib.c
rename to torchvision/csrc/io/image/cpu/giflib/dgif_lib.cpp
diff --git a/torchvision/csrc/io/image/cpu/giflib/gif_hash.c b/torchvision/csrc/io/image/cpu/giflib/gif_hash.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/gif_hash.c
rename to torchvision/csrc/io/image/cpu/giflib/gif_hash.cpp
diff --git a/torchvision/csrc/io/image/cpu/giflib/gifalloc.c b/torchvision/csrc/io/image/cpu/giflib/gifalloc.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/gifalloc.c
rename to torchvision/csrc/io/image/cpu/giflib/gifalloc.cpp
diff --git a/torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.c b/torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.c
rename to torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.cpp
-- 
2.43.0

